packages/nimble-components/src/table-column/text/index.ts

```diff
@@ -0,0 +1,54 @@
+import { attr } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import type { StringField } from '../../table/types';
+import { TableColumn } from '../base';
+import { styles } from '../base/styles';
+import { template } from '../base/template';
+import { cellStyles } from './styles';
+import { cellTemplate } from './template';
+
+export type TableColumnTextCellData = StringField<'value'>;
+export interface TableColumnTextColumnConfig {
+    placeholder: string;
+}
+
+/**
+ * The table column for displaying strings.
+ */
+export class TableColumnText extends TableColumn<
+TableColumnTextCellData,
+TableColumnTextColumnConfig
+> {
+    public cellStateDataFieldNames = ['value'] as const;
+
+    @attr({ attribute: 'field-name' })
+    public fieldName?: string;
+
+    @attr
+    public placeholder?: string;
+
+    public override readonly cellStyles = cellStyles;
+
+    public readonly cellTemplate = cellTemplate;
+
+    public getColumnConfig(): TableColumnTextColumnConfig {
+        return { placeholder: this.placeholder ?? '' };
+    }
+
+    public getRecordFieldNames(): string[] {
+        if (!this.fieldName) {
+            throw new Error('fieldName must be provided.');
+        }
```

Comment:
So, this might also mean that we need to change the signature of `getRecordFieldNames` to return a `(string | null | undefined)[]`. Throwing the error in the implementation as it does now allows us to avoid that. Thoughts?

Reply:
What about returning an empty array?

Reply:
Discussed off-line. We like returning `(string | undefined)[]` to simplify the implementation, as the column implementation doesn't need to concern itself with validation.